llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -304,13 +324,19 @@ void GCNSchedStrategy::initCandidate(SchedCandidate &Cand, SUnit *SU,
 
   int SGPRDelta = NewSGPRPressure - SGPRCriticalLimit;
   int VGPRDelta = NewVGPRPressure - VGPRCriticalLimit;
+  int AGPRDelta = NewAGPRPressure - AGPRCriticalLimit;
 
-  if (SGPRDelta >= 0 || VGPRDelta >= 0) {
+  if (SGPRDelta >= 0 || VGPRDelta >= 0 || AGPRDelta >= 0) {
     HasHighPressure = true;
+    // Prioritize reducing the VGPRDelta if both are >= 0
     if (SGPRDelta > VGPRDelta) {
```

Comment:
Should we check also if `SGPRDelta > AGPRDelta`?

Reply:
So the AGPR pressure should be prioritized over SGPR pressure?

Reply:
I think so -- especially since we are not using an error margin for the AGPR limit -- but maybe we should check if AGPR pressure is strictly greater than these limits.